vars/tdr.groovy

```diff
@@ -0,0 +1,11 @@
+def buildEndToEndTests(int delay, String buildJob, String[] params) {
+    build(job: buildJob, parameters: params, quietPeriod: delay, wait: false)
+}
+
+def setBuildStatus(String status, String desc) {
```

Comment:
How many build statuses are there?

If there are just two (success/failure), then would it be helpful to have two functions, e.g. `reportSuccessfulBranchBuild` or `reportFailedBranchBuild`? Then callers of the function don't have to work out what the possible status strings have to be.

Also, if the description only makes sense in the failure context, you only have to add the description parameter to the second function.

Reply:
I hadn't thought about doing it that way, I'll add a second function and make them a bit more specific!